pkg/infra/iac2/templates/security_group/factory.ts

```diff
@@ -0,0 +1,46 @@
+import * as aws from '@pulumi/aws'
+
+interface Args {
+    Name: string
+    Vpc: aws.ec2.Vpc
+}
+
+function create(args: Args): aws.ec2.SecurityGroup {
+    return new aws.ec2.SecurityGroup(args.Name, {
+        name: args.Name,
+        vpcId: args.Vpc.id,
+        egress: [
+            {
+                cidrBlocks: ['0.0.0.0/0'],
+                description: 'Allows all outbound IPv4 traffic.',
+                fromPort: 0,
+                protocol: '-1',
+                toPort: 0,
+            },
+        ],
+        ingress: [
+            {
+                description:
+                    'Allows inbound traffic from network interfaces and instances that are assigned to the same security group.',
+                fromPort: 0,
+                protocol: '-1',
+                self: true,
+                toPort: 0,
+            },
+            {
```

Comment:
These should be defined in the provider and configurable. We dont need to add the eks control plane rule unless we have eks in the architecture